test/rosdep_repo_check/make_rule.py

```diff
@@ -86,41 +86,55 @@ def key(self) -> Optional[str]:
             self._key = self._calculate_key()
         return self._key
 
-    def as_dict(self, minimal=True):
+    def as_dict(self):
         if self.key is None:
             raise ValueError('Rule has no key')
         yaml_dict = {}
 
-        # First pass, put all the data in
         for os, releases in self._platforms.items():
             if os not in yaml_dict:
                 yaml_dict[os] = {}
             for release in releases:
-                yaml_dict[os][release] = [self._packages[(os, release)]]
-
-        # Second pass, minimize keys
-        if minimal:
-            for os, release_packages in yaml_dict.items():
-                package_names = set([l[0] for l in release_packages.values()])
-                if len(package_names) == 1:
-                    # All releases have the same package name
-                    yaml_dict[os] = [package_names.pop()]
-                elif len(package_names) == 2 and None in package_names:
-                    # Not all releases have the package.
-                    # The ones that do have the same name.
-                    # This assumes releases without the package are older, and
-                    # may be the wrong thing to do if a distro stopped
-                    # releasing a package.
-                    package_name = [n for n in package_names if n][0]
-                    new_os_dict = {'*': [package_name]}
-                    for release, package in release_packages.items():
-                        if package is None:
-                            new_os_dict[release] = None
-                    yaml_dict[os] = new_os_dict
-    
+                package = self._packages[(os, release)]
+                yaml_dict[os][release] = [package] if package else None
+
         return {self.key: yaml_dict}
 
 
+def _minimize_rules(yaml_dict, supported_versions):
+    # 1. Remove entries which are entirely null
+    for key in list(yaml_dict.keys()):
+        for os in list(yaml_dict[key].keys()):
+            for release, package in (yaml_dict[key][os] or {}).items():
+                if package is not None:
+                    break
+            else:
+                del yaml_dict[key][os]
+        if not yaml_dict[key]:
+            del yaml_dict[key]
```

Comment:
This whole section can be dropped in light of https://github.com/ros/rosdistro/pull/33916#issuecomment-1191981410

```suggestion
```